src/inmanta/server/services/environmentservice.py

```diff
@@ -327,6 +351,17 @@ def get_depended_by(self) -> List[str]:
             result: ReturnValue[None] = ReturnValue(response=None)
             if warnings:
                 result.add_warnings(warnings)
+            await self.notify_listeners(EnvironmentAction.updated, env.to_dto())
             return result
         except KeyError:
             raise NotFound()
+
+    def register_listener(self, action: EnvironmentAction, listener: EnvironmentListener) -> None:
```

Comment:
I would document that this method should only be called during pre-start, as it may cause a race condition with the for-loop in notify_listeners when used while events are being processed